doc/pages/architecture/060-triggers.md

```diff
@@ -1,23 +1,376 @@
+This specific `condition` specifies that when data is received on the `org.astarteplatform.Values`
+interface on `/realValue` path, if the value of said data is `> 0.4`, then the trigger is activated.
+For more information about all the possible conditions, check out the [Conditions section](#conditions)
 
-Given this kind of flexibility, triggers are the most powerful way to push data to an external service, potentially without any additional customization.
+The `action` object describes what the result of the trigger will be. In this specific case, an HTTP
+`POST` request will be sent to `https://example.com/my_hook`, with the payload:
+
+```json
+{
+  "timestamp": "<event_timestamp>",
+  "device_id": "<device_id>",
+  "event": {
+    "type": "incoming_data",
+    "interface": "org.astarteplatform.Values",
+    "path": "/realValue",
+    "value": <some_value>
+  }
+}
+```
+
+To know more about all possible actions, check the [Actions section](#actions)
+
+## Conditions
+
+A condition defines the event upon which an action is triggered. Conditions are expressed through
+simple triggers. Astarte monitors incoming events and triggers a corresponding action whenever there
+is a match.
+
+Simple triggers are divided into two types: [Device Triggers](#device-triggers) and [Data
+Triggers](#data-triggers).
+
+### Device Triggers
+
+Device triggers express conditions matching the state of a device.
+
+This is the generic representation of a Data Trigger:
+
+```json
+{
+  "type": "device_trigger",
+  "on": "<device_trigger_type>",
+  "device_id": "<device_id>"
+}
+```
+
+#### Parameters
+
+`device_trigger_type` can be one of the following:
+- `device_connected`: triggered when a device connects to its transport.
+- `device_disconnected`: triggered when a device disconnects from its transport.
+
+`device_id` can be a specific Device ID or `*`, meaning the trigger will be installed for all
+devices in a Realm.
+
+### Data Triggers
+
+Data triggers express conditions matching data coming from a device.
+
+This is the generic representation of a Device Trigger:
+
+```json
+{
+  "type": "data_trigger",
+  "on": "<data_trigger_type>",
+  "interface_name": "<interface_name>",
+  "interface_major": "<interface_major>",
+  "match_path": "<match_path>",
+  "value_match_operator": "<value_match_operator>",
+  "known_value": <known_value>
+}
+```
+
+Data triggers are installed for all devices in a Realm.
+
+#### Parameters
+
+`data_trigger_type` can be one of the following:
+- `incoming_data`: verifies the condition whenever new data arrives.
+- `value_stored`: verifies the condition whenever new data arrives, after it is saved to the
+  database.
+- `value_change`: verifies the condition whenever the last sample received is different from the
```

Comment:
the last data or last value. value_change is also valid for properties, so we don't use the word sample.